### ISSUE TITLE
'break' in switch() case flags later cases as "unreachable"

### ISSUE THREAD

**Issue by MichaelChirico** [r-lib/lintr#3023]
Related: currently this is flagged as a lint:

```r
repeat {
  switch(x,
    a = break,
    b = "d"
  )
}
```

```
<text>:4:14: warning: [unreachable_code_linter] Remove code and comments coming after `break`.
    a = break,
             ^
```

**Comment by MichaelChirico** [comment]
#3084 has the same apparent root cause since #3085 fixes both
